### ISSUE TITLE
Xwayland is incompatible with KISS Mesa and misses dependencies

### ISSUE THREAD

**Issue by apprehensions** [ehawkvu/kiss-xorg#136]
If a regular user were on KISS, and decided to add this repository to `KISS_PATH` for the purpose of compiling Xwayland, they will be hit with this error when compiling Xwayland:

```
glx/meson.build:43:12: ERROR: Dependency "gl" not found, tried pkgconfig and system
```

This is because KISS Mesa does not compile GLX, and has X11 disabled. What i propose:

+ a custom `xwayland` or `wayland` directory with `xwayland` and `mesa` that is made to compile for both X11 and wayland.

Another thing is that when launching Xwayland, it has the run-time dependency of `xkbcomp`, which is NOT listed as a dependency for the package.

**Comment by ehawkvu** [comment]
Alternatively we could just add a check for wayland in the mesa package and enable wayland support on the fly. I've noted this issue in the past in #101 . I'll fix the xkbcomp issue.

**Comment by ehawkvu** [comment]
I'm not totally understanding your use case, since what I suggested prior would solve the new user problem of using Xwayland. Maybe I'm not understanding the particulars of why one would need this setup?

**Comment by apprehensions** [comment]
> I'll fix the xkbcomp issue.

you forgot

**Comment by ehawkvu** [comment]
xkbcomp resolved as of 8eb8e843b393ce30eda19488d09cba9eddbb280c
